File: backend/src/FarmManagement.Application/DTOs/LaborActivities/LaborActivityModels.cs
namespace FarmManagement.Application.DTOs.LaborActivities;

public sealed record NamedReferenceResponse(Guid Id, string Name);

public sealed record LaborActivityResponse(
    Guid Id,
    DateOnly ActivityDate,
    NamedReferenceResponse Farm,
    NamedReferenceResponse? FarmArea,
    NamedReferenceResponse? Plantation,
    NamedReferenceResponse? CropCycle,
    NamedReferenceResponse ActivityType,
    string Status,
    string? Description,
    string? CancellationReason,
    DateTimeOffset CreatedAt,
    DateTimeOffset? UpdatedAt);

public sealed record CreateLaborActivityRequest(
    DateOnly? ActivityDate,
    Guid? FarmId,
    Guid? FarmAreaId,
    Guid? PlantationId,
    Guid? CropCycleId,
    Guid? LaborActivityTypeId,
    string? Description,
    int? WorkerCount,
    decimal? TotalWorkingHours,
    decimal? CostAmount,
    string? Status);

public sealed record UpdateLaborActivityRequest(
    DateOnly? ActivityDate,
    Guid? FarmId,
    Guid? FarmAreaId,
    Guid? PlantationId,
    Guid? CropCycleId,
    Guid? LaborActivityTypeId,
    string? Description,
    int? WorkerCount,
    decimal? TotalWorkingHours,
    decimal? CostAmount,
    string? Status);

public sealed record CancelLaborActivityRequest(string? Reason);

public sealed record LaborActivityActor(Guid UserId, Guid OrganizationId);

File: frontend/farm-management-web/src/app/features/labor-activities/models/labor-activity.models.ts
import { PagedResponse } from "../../../core/models/paged-response.model";

export interface NamedReference {
  readonly id: string;
  readonly name: string;
}

export type LaborActivityStatus = "DRAFT" | "COMPLETED" | "CANCELLED";

export interface LaborActivity {
  readonly id: string;
  readonly activityDate: string;
  readonly farm: NamedReference;
  readonly farmArea: NamedReference | null;
  readonly plantation: NamedReference | null;
  readonly cropCycle: NamedReference | null;
  readonly activityType: NamedReference;
  readonly workerCount: number;
  readonly totalWorkingHours: number | null;
  readonly costAmount: number | null;
  readonly currency: string;
  readonly status: LaborActivityStatus;
  readonly description: string | null;
  readonly createdAt: string;
  readonly updatedAt: string | null;
}

export type LaborActivityList = PagedResponse<LaborActivity>;

export interface LaborActivityFilter {
  readonly farmId?: string;
  readonly farmAreaId?: string;
  readonly plantationId?: string;
  readonly cropCycleId?: string;
  readonly activityTypeId?: string;
  readonly fromDate?: string;
  readonly toDate?: string;
  readonly status?: string;
  readonly page: number;
  readonly pageSize: number;
}

export interface CancelLaborActivityRequest {
  readonly reason: string;
}
